src/libraries/Pool.sol

```diff
@@ -414,17 +414,22 @@ library Pool {
                     );
                     // if we're moving leftward, we interpret liquidityNet as the opposite sign
                     // safe because liquidityNet cannot be type(int128).min
-                    unchecked {
-                        if (params.zeroForOne) liquidityNet = -liquidityNet;
-                    }
+                    liquidityNet = flipLiquidityDelta(liquidityNet, params.zeroForOne);
 
                     state.liquidity = liquidityNet < 0
                         ? state.liquidity - uint128(-liquidityNet)
                         : state.liquidity + uint128(liquidityNet);
                 }
 
+                // Equivalent to `state.tick = zeroForOne ? step.tickNext - 1 : step.tickNext;`
                 unchecked {
-                    state.tick = params.zeroForOne ? step.tickNext - 1 : step.tickNext;
```

Comment:
![Screenshot 2024-04-08 at 3 18 32 PM](https://github.com/Uniswap/v4-core/assets/34962750/6fe8a802-8127-4cb6-8f47-764bd0a3b734)

I ran this equivalence locally and gas doesnt go down for it, it goes up. Notably the `simple swap` gas tests go up by 100, as do other swap tests